### ISSUE TITLE
transclusion inherit tags instead of inheriting title-as-tag (or copy-pasta tags en masse)

### ISSUE THREAD

**Issue by rcrath** [Jermolene/TiddlyWiki5#3215]
TW 5.16, nodejs version, linux, chrome browser.

Any way to already do the desired feature?  Otherwise, this is a feature request.

## Current behavior
- transclude text
- option to tag transcluded tiddler with inherited title.

## desired behavior
- transclude text
- have option to inherit tags

## rationale:
I have a pretty thick and useful tagging system.  When I transclude stuff in order to keep tiddler size manageable, I almost always want the transcluded text to contain the same tags.  I understand that "inherit title" links back to the tiddler with all the tags, but I have kind of meta tags, like the name of the project, that go with all the tiddlers in that project.  With inherited titles, I lose the connection to the project unless I tag each transcluded tiddler manually.  That would not be so bad, but when tracking things down, I find having all the tags, often half a dozen or more, to be useful so I don't have to navigate extra hierachy.  If I break up one big tiddler with eight tags into many small ones, that is a lot of manual tagging.

Example, 

Recently wrote a page on a particular African American Spiritual, "Knee-bone."  It inherited its tags via [this hack](https://groups.google.com/forum/#!searchin/TiddlyWiki/rich.rath@gmail.com$20tags|sort:date/tiddlywiki/11Ju50q-nHQ/9gtdW8-zCgAJ) from the parent, which is a reference to the book I found "Knee-bone" in.  The reference book tiddler had the following tags which the "Knee-bone" tiddler inherited:  19th c ×  African American ×  creolization ×  hearing ×  labor ×  music ×  race ×  Sea Islands ×  slavery ×.  When I was done with the "Knee-bone" tiddler, I had my analysis, quoted text, the book that it came from, lyrics, and sheet music from imported image tiddlers.  

It was a long page, so I transcluded lyrics and sheet music as new tiddlers with links from "Knee-bone."  Having the tag be "Knee-bone" drops it out of all the other tagging contexts.  I don't always arrive at pages taxonomically because the wiki is pretty big (I've been building it for quite a few years!), so while searching on the tags would bring me to the parent tiddler, I'd rather have a "flatter but thicker" tagging set that does not create hierachy even at the cost of repetitive tags.  Takes some extra clicks out of the equation.

Another possible way of going about it would be to be able to copy and past a whole set of tags.  Right now they seem like they are pretty atomic nd "uncopypastable" in that regard.

I'm hoping that maybe a small tweak to the google groups code would do it, but I'm not much of a coder at all, so I am not able to figure it out myself.  

Others might find it useful enough to make it an option though, you think?

**Comment by pmario** [comment]
Let me formulate it a bit different, to see, if I did understand your issue. 

There is a book: 

>title: BOOK
tags: [[19th c]] [[African American]] creolization hearing labor music race Sea Islands slavery 

There is a tiddler: 

>title: Knee-bone
tags: [[19th c]] [[African American]] creolization hearing labor music race Sea Islands slavery 


There are 2 tiddlers that are transcluded into "Knee-bone"

>title: lyrics
tags: Knee-bone

>title: music
tags: Knee-bone

Now there should be simple workflow to tag the tiddlers "music" and "lyrics" with the same tags from 

Knee-bone or BOOK?

Is that right?

**Comment by pmario** [comment]
If my assumptions are right, you basically need a simple way to "copy-paste" tags. right

**Comment by rcrath** [comment]
yes that is it.  I redid the tags on knee-bone manually, otherwise the only "correctly tagged" one (in my proposed tiddlyverse) would be "BOOK."  Copy pasting the tags would be one way, and that would be a generally useful feature, but I have a new tiddler button that grabs inherited tags (see link above) from whatever tiddler is focused when I open the new one.  I would like to do the same thing for transcluded pages where instead of inheriting the title-as-tag they would inherit the tags.  Copy paste of tags is one way, the button method would be another, but I am not sure where to start to figure out how to do either as far as js goes.

**Comment by rcrath** [comment]
Kneebone was transcluded as it got bigger from BOOK and the other two were transcluded from knee-bone as it got too big.

**Comment by buggyj** [comment]
If you clone you original tiddler you get its tags (and content), and you could delete the part of the content to form your new tiddler.
